src/test/suite/MockingUtils.ts

```diff
@@ -1,10 +1,30 @@
-import { PromiseWithChild } from 'child_process';
+
 import * as vscode from 'vscode';
 import { ConfigurationConstants, LanguageServerConstants } from '../../constants';
-import { ExecAsyncType, ExecOutput } from '../../language/dafnyInstallation';
 import { MockedOutputChannelBuilder } from './MockedOutputChannelBuilder';
 import { MockedWorkspace } from './MockedWorkspace';
 
+export interface ExecOutput {
+  stdout: string;
+  stderr: string;
+}
+export class MockingExec {
```

Comment:
Have you looked at using sinon for creating the stubs? I was hoping we can create a stub based on the original type.

Reply:
I just had a look at stubs, and it looks great, I'm just not sure about where to use it best.

In my first look, the version I found does not care about types being stubbed at all, it's just a wrapper around the JavaScript.
Also, I did not find a way it would support my use case for exec:

    public constructor() {
      this.stub = (command: string) => {
        throw `Unexpected comand ${command}`;
      };
    }

The best alternative I have found that sinon offers is nowhere as good:

    public constructor() {
      this.stub = sinon.stub().throwsArg()
    }

which throws the first argument. If I want to do anything more complicated than that, there are no other options. Writing a custom function is a better alternative to `sinon.stub()` in this case.
But that will only throw command, not a customized error message.

My conclusion is that, at this stage of development, I don't see a clear use case, so I would rather wait for the tests to become more complex, or working with someone else who has a clearer vision on how to use them.